Portal/src/Datahub.Infrastructure/Services/ProjectUserManagementService.cs

```diff
@@ -150,9 +167,60 @@ public async Task RemoveUserFromProject(string projectAcronym, string userGraphI
 
     }
 
-    public Task UpdateUserInProject(string projectAcronym, Datahub_Project_User user)
+    public async Task UpdateUserInProject(string projectAcronym, ProjectMember projectMember)
     {
-        throw new NotImplementedException();
+        _logger.LogInformation("Preparing to remove user {UserGraphId} from project {ProjectAcronym}", projectMember.UserId,
+            projectAcronym);
+
+        await using var context = await _contextFactory.CreateDbContextAsync();
+        var project = await context.Projects
+            .Include(p => p.Users)
+            .FirstOrDefaultAsync(p => p.Project_Acronym_CD == projectAcronym);
+
+        if (project == null)
+        {
+            _logger.LogError("Project {ProjectAcronym} not found when trying to remove user {UserGraphId}",
+                projectAcronym,
+                projectMember.UserId);
+            throw new ProjectNotFoundException($"Project {projectAcronym} not found");
+        }
+
+        var exists = await context.Project_Users
+            .FirstOrDefaultAsync(u => u.User_ID == projectMember.UserId && u.Project.Project_ID == project.Project_ID);
+
+        if (exists is null)
+        {
+            _logger.LogInformation("User {UserGraphId} does not exist in project {ProjectAcronym}", projectMember.UserId,
+                projectAcronym);
+            throw new UserNotFoundException($"User {projectMember.UserId} not found in project {projectAcronym}");
+        }
+
+        if (!ShouldUpdateUser(exists, projectMember.Role))
+        {
+            _logger.LogInformation("User {UserGraphId} already has role {Role} in project {ProjectAcronym}", projectMember.UserId,
+                projectMember.Role, projectAcronym);
+            return;
+        }
+        
+        using var scope = new TransactionScope(
+            TransactionScopeOption.Required, TransactionScopeAsyncFlowOption.Enabled);
+
+        try
+        {
+            exists.IsDataApprover = projectMember.Role == ProjectMemberRole.Publisher;
+            exists.IsAdmin = exists.IsDataApprover || projectMember.Role == ProjectMemberRole.Admin;
```

Comment:
This line seems suspicious, is an admin and a data approver the same?

Reply:
Yes, this is the current structure of the model. @yjmrobert should we update the model to use the new enum?

Reply:
Makes sense to me